## 단원별 심화 연습 문제

In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import seaborn as sns
import glob

# set floating point formatting
pd.options.display.float_format = '{:,.6f}'.format

### Q3

**범위**
- (이전 범위 포함)
- 결측치 채우기

In [7]:
import seaborn as sns

titanic = sns.load_dataset('titanic')
titanic.head()

survived  pclass     sex       age  sibsp  parch      fare embarked  class  \
0         0       3    male 22.000000      1      0  7.250000        S  Third   
1         1       1  female 38.000000      1      0 71.283300        C  First   
2         1       3  female 26.000000      0      0  7.925000        S  Third   
3         1       1  female 35.000000      1      0 53.100000        S  First   
4         0       3    male 35.000000      0      0  8.050000        S  Third   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False  
2  woman       False  NaN  Southampton   yes   True  
3  woman       False    C  Southampton   yes  False  
4    man        True  NaN  Southampton    no   True

각 **컬럼별 결측치를 출력**하세요

In [3]:
# 코드를 입력해 주세요
titanic.isna().sum()

survived         0
pclass           0
sex              0
age            177
sibsp            0
parch            0
fare             0
embarked         2
class            0
who              0
adult_male       0
deck           688
embark_town      2
alive            0
alone            0
dtype: int64

<p><strong>[출력 결과]</strong></p><pre>survived         0
pclass           0
sex              0
age            177
sibsp            0
parch            0
fare             0
embarked         2
class            0
who              0
adult_male       0
deck           688
embark_town      2
alive            0
alone            0
dtype: int64</pre>

`age` 컬럼의 결측 데이터를 다음의 조건에 맞도록 채워 주세요

- `who`가 `man`인 데이터는 결측치가 아닌 `age` 컬럼의 **median**으로 결측치를 채워주세요
- `who`가 `woman`인 데이터는 결측치가 아닌 `age` 컬럼의 **25% Quantile 값**으로 결측치를 채워주세요
- `who`가 `child`인 데이터는 결측치가 아닌 `age` 컬럼의 **평균**으로 결측치를 채워주세요

In [8]:
# 코드를 입력해 주세요
cond1 = (titanic['who'] == 'man')
cond1_age = titanic.loc[cond1, 'age'].median()
titanic.loc[cond1, 'age'] = titanic.loc[cond1, 'age'].fillna(cond1_age)

cond2 = (titanic['who'] == 'woman')
cond2_age = titanic.loc[cond2, 'age'].quantile(0.25)
titanic.loc[cond2, 'age'] = titanic.loc[cond2, 'age'].fillna(cond2_age)

cond3 = (titanic['who'] == 'child')
cond3_age = titanic.loc[cond3, 'age'].mean()
titanic.loc[cond3, 'age'] = titanic.loc[cond3, 'age'].fillna(cond3_age)

titanic['age'].isnull().sum()

0

In [9]:
# 검증코드
print(f"결측치: {titanic['age'].isnull().sum()}")
print(f"age mean: {titanic['age'].mean():.4f}")

결측치: 0
age mean: 29.3425


<p><strong>[출력 결과]</strong></p><pre>결측치: 0
age mean: 29.3425
</pre>

## 제출

제출을 위해 새로 로드된 타이타닉 데이터셋에서 `age` 컬럼의 결측치를 다음 조건에 맞춰 채운 결과를 `result_df`에 저장하세요.

- `who`가 `man`인 데이터는 결측치가 아닌 `age` 컬럼의 **median**으로 결측치를 채워주세요
- `who`가 `woman`인 데이터는 결측치가 아닌 `age` 컬럼의 **25% Quantile 값**으로 결측치를 채워주세요
- `who`가 `child`인 데이터는 결측치가 아닌 `age` 컬럼의 **평균**으로 결측치를 채워주세요

In [10]:
titanic = sns.load_dataset('titanic')

cond1 = (titanic['who'] == 'man')
cond1_age = titanic.loc[cond1, 'age'].median()
titanic.loc[cond1, 'age'] = titanic.loc[cond1, 'age'].fillna(cond1_age)

cond2 = (titanic['who'] == 'woman')
cond2_age = titanic.loc[cond2, 'age'].quantile(0.25)
titanic.loc[cond2, 'age'] = titanic.loc[cond2, 'age'].fillna(cond2_age)

cond3 = (titanic['who'] == 'child')
cond3_age = titanic.loc[cond3, 'age'].mean()
titanic.loc[cond3, 'age'] = titanic.loc[cond3, 'age'].fillna(cond3_age)

result_df = titanic.copy()

## 채점 수행

아래 코드는 채점 수행을 위한 코드입니다.

따라서 이를 수정했을시 **채점이 제대로 이루어지지 않습니다.**

**주의**: 채점 코드를 실행하기 전에 반드시 코드 파일을 한번 저장하시길 바랍니다.

In [11]:
import os

result_df.to_json('result.json')

os.system('elice_grade result.json cds_ds_exercise.ipynb')

send files ['result.json', 'cds_ds_exercise.ipynb'] for grade...
waiting result...
waiting result...
waiting result...
done!

Score: 100.000000
Duration: 2.837 seconds
=== Message ===
제출 완료되었습니다.


0